src/components/utils/popover/popover-abstract.ts

```diff
@@ -0,0 +1,318 @@
+  }
+
+  /**
+   * Adds custom html content to the popover
+   *
+   * @param content - html content to append
+   */
+  private addCustomContent(content: HTMLElement): void {
+    this.nodes.customContent = content;
+    this.nodes.customContent.classList.add(css.customContent);
+    this.nodes.popoverContainer?.insertBefore(content, this.nodes.popoverContainer.firstChild);
+  }
+
+  /**
+   * Handles clicks inside popover
+   *
+   * @param event - item to handle click of
+   */
+  private handleClick(event: Event): void {
+    const item = this.getTargetItem(event);
+
+    if (item === undefined) {
+      return;
+    }
+
+    if (item.isDisabled) {
+      return;
+    }
+
+    if (item.children.length > 0) {
+      this.handleShowingNestedItems(item);
+
+      return;
+    }
+
+    /** Cleanup other items state */
+    this.items.filter(x => x !== item).forEach(x => x.reset());
+
+    item.handleClick();
+
+    this.toggleItemActivenessIfNeeded(item);
+
+    if (item.closeOnActivate) {
+      this.hide();
+    }
+  }
+
+  /**
+   * Toggles nothing found message visibility
+   *
+   * @param isDisplayed - true if the message should be displayed
+   */
+  private toggleNothingFoundMessage(isDisplayed: boolean): void {
+    this.nodes.nothingFoundMessage?.classList.toggle(css.nothingFoundMessageDisplayed, isDisplayed);
+  }
+
+  /**
+   * Toggles custom content visibility
+   *
+   * @param isDisplayed - true if custom content should be displayed
+   */
+  private toggleCustomContent(isDisplayed: boolean): void {
+    this.nodes.customContent?.classList.toggle(css.customContentHidden, isDisplayed);
+  }
+
+  /**
+   * - Toggles item active state, if clicked popover item has property 'toggle' set to true.
+   *
+   * - Performs radiobutton-like behavior if the item has property 'toggle' set to string key.
+   * (All the other items with the same key get inactive, and the item gets active)
+   *
+   * @param clickedItem - popover item that was clicked
+   */
+  private toggleItemActivenessIfNeeded(clickedItem: PopoverItem): void {
+    if (clickedItem.toggle === true) {
+      clickedItem.toggleActive();
+    }
+
+    if (typeof clickedItem.toggle === 'string') {
+      const itemsInToggleGroup = this.items.filter(item => item.toggle === clickedItem.toggle);
+
+      /** If there's only one item in toggle group, toggle it */
+      if (itemsInToggleGroup.length === 1) {
+        clickedItem.toggleActive();
+
+        return;
+      }
+
+      /** Set clicked item as active and the rest items with same toggle key value as inactive */
+      itemsInToggleGroup.forEach(item => {
+        item.toggleActive(item === clickedItem);
+      });
+    }
+  }
+
+  /**
+   * Handles displaying nested items for the item. Behaviour differs depending on platform.
+   *
+   * @param item – item to show nested popover for
+   */
+  protected abstract handleShowingNestedItems(item: PopoverItem): void;
```

Comment:
name and description are a bit confusing. What does it mean "handle displaying" or "handle showing"? Maybe something like `showNestedListIfNeeded`?

Reply:
To handle showing nested items means to do what is necessary to make them visible. This behaviour differs depending on platform: on desktop nested popover is displayed and on mobile nested items are rendered in the same popover

Reply:
why not just `showNestedItems()?`